instrumenting/clientlibs: add process_threads (#1964)
prometheus/docs: content/docs/instrumenting/writing_clientlibs.md

| `process_resident_memory_bytes`    | Resident memory size in bytes.                         | bytes            |
| `process_heap_bytes`               | Process heap size in bytes.                            | bytes            |
| `process_start_time_seconds`       | Start time of the process since unix epoch in seconds. | seconds          |

> | `process_threads`                  | Number of OS threads that currently exist.             | threads          |

Reviewer: "currently exist" is a bit ambiguous. How about this:

```suggestion
| `process_threads`                  | Number of OS threads in the process.             | threads          |
```

Author: I based this on the Golang goroutines gauge [description](https://github.com/prometheus/client_golang/blob/v1.11.0/prometheus/go_collector.go#L47), but yes I prefer your wording too.
